clippy_lints/src/slow_vector_initialization.rs

```diff
@@ -0,0 +1,330 @@
+// Copyright 2014-2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::rustc::hir::intravisit::{walk_expr, NestedVisitorMap, Visitor};
+use crate::rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use crate::rustc::{declare_tool_lint, lint_array};
+use crate::rustc::hir::*;
+use if_chain::if_chain;
+use crate::syntax_pos::symbol::Symbol;
+use crate::syntax::ast::{LitKind, NodeId};
+use crate::syntax::source_map::Span;
+use crate::utils::{match_qpath, span_lint_and_then, SpanlessEq};
+use crate::utils::get_enclosing_block;
+use crate::rustc_errors::{Applicability};
+
+/// **What it does:** Checks slow zero-filled vector initialization
+///
+/// **Why is this bad?** This structures are non-idiomatic and less efficient than simply using
+/// `vec![len; 0]`.
```

Comment:
Shouldn't this be...
```suggestion
/// `vec![0; len]`.
```

Reply:
Yes, I swapped the arguments. Thanks!

Reply:
It could be worth a new lint, though.
There is no point in using `vec![x; 0]` because it's an empty array, [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=10eb608b5441de56440a5dcf2569ecf2).

Reply:
I'm talking about new lint which would catch mistakes like that.